app/lib/hooks/useUserPermissions.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+
+import { TSubscriptionModel } from '../../definitions';
+import { hasPermission } from '../methods/helpers';
+import { useAppSelector } from './useAppSelector';
+
+interface IMembersSection {
+	rid: TSubscriptionModel['rid'];
+	t: TSubscriptionModel['t'];
+	joined: boolean;
+}
+// Promise.allSettled() is not working for react native, in the future use this instead of various useEffects
+// https://github.com/facebook/react-native/issues/30236#issuecomment-939286987
+export const useUserPermissions = ({ rid, t, joined }: IMembersSection): { canAddUser: boolean; canInviteUser: boolean } => {
```

Comment:
I don't think we need this custom hook.
We have `usePermissions` for this use case.

Also, this custom hooks wouldn't be reactive.